packages/material-ui/src/styles/createTypography.js

```diff
@@ -27,7 +28,7 @@ export default function createTypography(palette, typography) {
     // Tell Material-UI what's the font-size on the html element.
     // 16px is the default font-size used by browsers.
     htmlFontSize = 16,
-    suppressDeprecationWarnings = process.env.MUI_SUPPRESS_DEPRECATION_WARNINGS,
+    suppressDeprecationWarnings = getEnv('MUI_SUPPRESS_DEPRECATION_WARNINGS'),
```

Comment:
`suppressDeprecationWarnings` is only used in the Typography component. moving `MUI_SUPPRESS_DEPRECATION_WARNINGS` there can help with code splitting.